### PR TITLE
Copy github.com/decred/dcrutil/bloom to bloom package.

### PR DESCRIPTION
**Pull request by jrick**
Use this new bloom package rather than relying on the bloom package
from the dcrutil repo.

Some minor source code changes have been made to conform to the dcrd
linter requirements.

### CODE REVIEW

File: bloom/example_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2014-2016 The btcsuite developers
+// Copyright (c) 2015-2016 The Decred developers
```
Comment (davecgh) [inline bloom/example_test.go]: 2015-2017 since the file had to be modified.

File: bloom/merkleblock_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2013-2016 The btcsuite developers
+// Copyright (c) 2015-2016 The Decred developers
```
Comment (davecgh) [inline bloom/merkleblock_test.go]: 2017

File: bloom/filter_test.go [hunk 3]
```diff
@@ -0,0 +1,641 @@
+// Copyright (c) 2013-2016 The btcsuite developers
+// Copyright (c) 2015-2016 The Decred developers
```
Comment (davecgh) [inline bloom/filter_test.go]: 2017

File: bloom/murmurhash3_test.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2013, 2014 The btcsuite developers
+// Copyright (c) 2015 The Decred developers
```
Comment (davecgh) [inline bloom/murmurhash3_test.go]: 2015-2017

**Comment by jrick** [comment]
updated

**Pull request merged**